independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/ModelUtils.java

```diff
@@ -184,6 +210,39 @@ public static String getVersion(Model model) {
         throw new IllegalStateException("Failed to determine version for project model");
     }
 
+    public static String getVersion(Model model) {
+        final String rawVersion = getRawVersion(model);
+        return isUnresolvedVersion(rawVersion) ? resolveVersion(rawVersion, model) : rawVersion;
+    }
+
+    public static boolean isUnresolvedVersion(String version) {
+        return getUnresolvedVersionPattern().matcher(version).find();
+    }
+
+    public static String resolveVersion(String rawVersion, Model rawModel) {
+        final Map<String, String> props = new HashMap<>();
```

Comment:
this would not handle case where there are nested properties would it ?

Reply:
Nesting is not allowed in Maven project versions.